src/testUtils/index.tsx

```diff
@@ -79,20 +160,309 @@ export const getSpecWrapper = (
     componentWrapper: ReactWrapper<{}, {}> | ShallowWrapper<{}, {}>,
     specName: string,
     attributeName: string = 'data-spec'
-): any => {
-    return componentWrapper.find(`[${attributeName}="${specName}"]`);
-};
+): any => componentWrapper.find(`[${attributeName}="${specName}"]`);
 
-const DEBUG = false;
+export const composeDuxState = (
+    query: Query<AppState | Partial<AppState>> = {},
+    duxState = baseState
+) => mutate(duxState, query);
 
-export const timeStart = (name: string) => {
-    if (DEBUG) {
-        console.time(name);
-    }
-};
+export const composeComponentTestUtils = <P extends {}>(
+    Component: React.ComponentClass | React.SFC,
+    baseProps: P = {} as any,
+    baseDuxState: AppState | Partial<AppState> = baseState,
+    baseContext: ConfigProviderContext = configProviderContext
+) => ({
+    setup: composeSetup<P>(Component, baseProps, baseDuxState, baseContext),
+    spyOn: composeSpy(Component)
+});
 
-export const timeEnd = (name: string) => {
-    if (DEBUG) {
-        console.timeEnd(name);
-    }
-};
+/**
+ * Create a select control option
+ */
+export const genSelectOption = ({ label }: { label: string }) => ({
+    label: capitalize(label.trim()),
+    labelKey: 'name',
+    valueKey: 'id'
+});
+
+export const genSendMsgAction = ({
+    uuid = 'send_msg-0',
+    text = 'Hey!',
+    all_urns = false
+}: {
+    uuid?: string;
+    text?: string;
+    // tslint:disable-next-line:variable-name
+    all_urns?: boolean;
+} = {}): SendMsg => ({
+    type: Types.send_msg,
+    uuid,
+    text,
+    all_urns
+});
+
+export const genSendEmailAction = ({
```

Comment:
adding initialization methods for each component we're testing. i like that we can use default parameters to return 'default versions' of each item as opposed to keeping them nearby in a `__test__` file, but happy to remove if you think this is overkill 🤙